### PR TITLE
Group 4

### PR DESCRIPTION
**Pull request by maxschwinghammer**

### CODE REVIEW

File: src/main/java/com/auberer/compilerdesignlectureproject/ast/ASTForNode.java [hunk 1]
```diff
@@ -12,4 +12,21 @@ public <T> T accept(ASTVisitor<T> visitor) {
...
+    public ASTStmtLstNode getBody() {
+        return getChild(ASTStmtLstNode.class, 0);
+    }
+
+    public ASTAssignStmtNode getInitialization() {
+        return getChild(ASTAssignStmtNode.class, 0);
+    }
+
+    public ASTAssignStmtNode getCondition() {
+        return getChild(ASTAssignStmtNode.class, 1);
```
Comment (marcauberer) [inline src/main/java/com/auberer/compilerdesignlectureproject/ast/ASTForNode.java:26]: This isn't correct anymore. Should be a LogicalExpr now

File: src/test/java/com/auberer/compilerdesignlectureproject/parser/ForNodeIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        assertNotNull(body);
+        assertNotNull(initialization);
+        assertNotNull(condition);
+        assertNotNull(increment);
+    }
+
+    // Unit tests for ASTForNode methods
+    @Test
+    void testGetBody() {
+        ASTForNode forNode = new ASTForNode();
```
Comment (marcauberer) [inline src/test/java/com/auberer/compilerdesignlectureproject/parser/ForNodeIntegrationTest.java:43]: Theoretically, you can integrate the tests for the helpers into the parsing test above.

File: src/main/java/com/auberer/compilerdesignlectureproject/ast/ASTForNode.java (hunk 1 above)
Comment (tech-n1c) [reply src/main/java/com/auberer/compilerdesignlectureproject/ast/ASTForNode.java]: Thanks for letting us know, this should now be fixed everywhere.

File: src/test/java/com/auberer/compilerdesignlectureproject/parser/ForNodeIntegrationTest.java (hunk 2 above)
Comment (marcauberer) [reply src/test/java/com/auberer/compilerdesignlectureproject/parser/ForNodeIntegrationTest.java:41]: Acknowleged

**Review by marcauberer — APPROVED** [review]
Will merge this and change the increment expression to an assignstmt. Thank you!

**Pull request merged**